src/api/nft.ts

```diff
@@ -1,15 +1,69 @@
 import { NftMetadata, NftTokenType, TokenUri } from '../types/types';
-import { RawBaseNft, RawNft } from '../internal/raw-interfaces';
+import {
+  RawBaseNft,
+  RawBaseNftContract,
+  RawNft,
+  RawNftContract
+} from '../internal/raw-interfaces';
 import { BigNumber } from 'ethers';
 
+export interface NftContractMetadata {
+  /** The name of the contract. */
+  name: string;
+  /** The symbol of the contract. */
+  symbol: string;
+  /** The number of NFTs in the contract. */
+  totalSupply: number;
+  /** The type of the token in the contract. */
+  tokenType?: NftTokenType;
+}
+
 /**
- * Represents an NFT contract.
+ * Alchemy representation of a base NFT contract that doesn't contain metadata.
  *
  * @public
  */
-export interface NftContract {
-  /** The NFT contract address. */
+export class BaseNftContract {
```

Comment:
I think it'd be cleaner if these were just interfaces instead of classes, like `TokenUri`. Just for context, I originally had `Nft` and `BaseNft` as classes since I wanted to eventually add helper methods onto it. In this case, having a simple interface would be cleaner.

Also, as discussed, can we move contractMetadata fields up for better ergonomics? Here's what I'm roughly thinking
```ts
export interace BaseNftContract {
  address: string
}

export interface NftContract extends BaseNftContract {
  name?: string
  symbol?: string
  ...
}
```

Reply:
Can lift contractMetadata, no problem!

For the interface vs. classes, don't we need the class to implement "fromResponse" to parse from RawBaseNFTContract?

Reply:
Moved to utils.ts!